pip/src/interpreter.rs

```diff
@@ -535,3 +549,21 @@ fn get_dict_opt_string(dict: &PyDict, key: &str) -> PyResult<Option<String>> {
         None => None,
     })
 }
+fn get_dict_opt_list_string(dict: &PyDict, key: &str) -> PyResult<Vec<String>> {
+    let value = dict.get_item(key)?;
+    let list: &PyList = match value {
+        Some(item) => item.downcast::<PyList>()?,
+        None => return Ok(vec![]),
+    };
+    match list
+        .iter()
+        .map(|item| {
+            item.downcast::<PyString>()
+                .map(|s| s.to_string_lossy().into())
+        })
+        .collect::<std::result::Result<Vec<String>, _>>()
+    {
+        Ok(list) => Ok(list),
+        Err(e) => Err(QSharpError::new_err(e.to_string())),
```

Comment:
This error type doesn't look quite right to me, `QSharpError` is more for like compiler errors, but I'm getting a bit lost in the types... what's `e` here, and is there a more direct conversion? Do you need to convert at all? I remember doing a bunch of work in this file's error handling but none of it makes sense anymore...